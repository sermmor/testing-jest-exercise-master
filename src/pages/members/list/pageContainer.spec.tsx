import * as React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { MemberListPageContainer } from './pageContainer';

const getMockStore = configureStore();

describe('pages/members/list/pageContainer tests', () => {
    it('should be a valid pageContainer rendered', () => {
        const state = {
            members: [
                {
                    id: 1,
                    login: 'test field login',
                    avatar_url: 'http://testavatar.png',
                },
                {
                    id: 2,
                    login: 'test field login (1)',
                    avatar_url: 'http://testavatar(1).png',
                },
            ],
        };
    
        const store = getMockStore(state);
    
        const componentResult = shallow(<MemberListPageContainer />, { context: { store }, } );
        
        expect(componentResult).toMatchSnapshot();
    });
});