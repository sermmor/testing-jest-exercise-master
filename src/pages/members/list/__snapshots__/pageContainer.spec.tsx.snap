// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/pageContainer tests should be a valid pageContainer rendered 1`] = `
<PageContainer
  fetchMembers={[Function]}
  members={
    Array [
      Object {
        "avatarUrl": "http://testavatar.png",
        "id": 1,
        "name": "test field login",
      },
      Object {
        "avatarUrl": "http://testavatar(1).png",
        "id": 2,
        "name": "test field login (1)",
      },
    ]
  }
/>
`;
